umake/ui/cli/__init__.py

```diff
@@ -322,30 +315,34 @@ def main(parser):
                            report=lambda arg: fetch_package_url.set() if arg == 'all downloads finished' else None)
             fetch_package_url.wait()
             user_version = framework.get_current_user_version(install_path)
-            latest_version = framework.parse_latest_version_from_package_url()
+            latest_version = framework.get_latest_version()
             is_outdated = is_first_version_higher(latest_version, user_version) \
-                if latest_version != 'Missing information' and user_version != 'Missing information' else False
-            frameworks_version.append({
-                'framework_name': framework_name,
-                'category_name': category_name,
-                'user_version': user_version,
-                'latest_version': latest_version,
-                'is_outdated': is_outdated
-            })
-        pretty_print_package_manager_output(frameworks_version)
-        outdated_frameworks = [item for item in frameworks_version if item['is_outdated']]
-        if not outdated_frameworks:
+                if (latest_version is not None and user_version is not None) else False
+            if is_outdated:
+                outdated_frameworks.append({
+                    'framework_name': framework_name,
+                    'category_name': category_name,
+                    'user_version': user_version,
+                    'latest_version': latest_version,
+                    'is_outdated': is_outdated,
+                    'forbidden_to_update': framework.forbidden_to_update
+                })
+        if len(outdated_frameworks) == 0:
+            print('All packages are up-to-date.')
             sys.exit(0)
         else:
-            args = parser.parse_args([outdated_frameworks[0]['category_name'], outdated_frameworks[0]['framework_name']])
-            CliUI()
-            run_command_for_args(args)
-
-            return
+            pretty_print_versions(outdated_frameworks)
+            for outdated_framework in outdated_frameworks:
+                if outdated_framework['forbidden_to_update'] is False:
+                    args = parser.parse_args([outdated_framework['category_name'], outdated_framework['framework_name']])
+                    CliUI()
+                    run_command_for_args(args)
+                    return
```

Comment:
If multiple outdated frameworks are present, only the first framework updates (the return statement is early).
I struggled a bit with running CliUI() in a loop.

Reply:
I'll look into this.
The best way would probably be to relaunch umake, but I'll let you know